iroh-net/src/discovery/local_swarm_discovery.rs

```diff
@@ -210,7 +210,7 @@ impl LocalSwarmDiscovery {
             );
 
             sender
-                .send(Message::Discovery(node_id.to_string(), peer.clone()))
+                .send_blocking(Message::Discovery(node_id.to_string(), peer.clone()))
```

Comment:
Reading the docs this is pretty sad.  They say you should not call this from async contexts, but there is nothing we can do to make sure this function is not called from async contexts.  In fact this **is** called from an async context because it spawns a new task.

Maybe you can convert this function into async without too many repercussions?

Reply:
I would prefer to keep this PR solely a 1:1 flume -> async_channel replacement. Should I make an issue?